_functions_exo_docker.sh

```diff
@@ -0,0 +1,210 @@
+  if ! ${DEPLOY_EXO_DOCKER}; then
+    return;
+  fi
+  env_var DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME "${INSTANCE_KEY}_exo"
+}
+
+#
+# Drops all exo docker data used by the instance.
+#
+do_drop_exo_docker_data() {
+  echo_info "Dropping exo docker data ..."
+  if ${DEPLOY_EXO_DOCKER}; then
+    echo_info "Drops Exo docker container ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME} ..."
+    delete_docker_container ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME}
+    echo_info "Drops Exo docker volume ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME}_codec..."
+    delete_docker_volume ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME}_codec
+    echo_info "Drops Exo docker volume ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME}_data ..."
+    delete_docker_volume ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME}_data
+    echo_info "Done."
+    echo_info "exo docker data dropped"
+  else
+    echo_info "Skip Drops exo docker container ..."
+  fi
+}
+
+do_create_exo_docker() {
+  if ${DEPLOY_EXO_DOCKER}; then
+    echo_info "Creation of the Exo docker volume ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME}_codec ..."
+    create_docker_volume ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME}_codec
+    echo_info "Creation of the Exo docker volume ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME}_data ..."
+    create_docker_volume ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME}_data    
+  fi
+}
+
+do_configure_exo_docker_ldap(){
+  if [ "${DEPLOYMENT_LDAP_ENABLED}" == "true" ]; then
+    echo_info "Start Deploying Directory ${USER_DIRECTORY} conf ..."      
+    mkdir -p ${DEPLOYMENT_DIR}/gatein/conf/portal/portal
+    env_var LDAP_HOST "ldap"
+    env_var DEPLOYMENT_LDAP_PORT "389"
+    cp ${ETC_DIR}/gatein/portal/portal/configuration.xml ${DEPLOYMENT_DIR}/gatein/conf/portal/portal/configuration.xml
+    evaluate_file_content ${ETC_DIR}/gatein/portal/portal/idm-configuration.xml.template ${DEPLOYMENT_DIR}/gatein/conf/portal/portal/idm-configuration.xml
+    evaluate_file_content ${ETC_DIR}/gatein/portal/portal/picketlink-idm-${USER_DIRECTORY}-config.xml.template ${DEPLOYMENT_DIR}/gatein/conf/portal/portal/picketlink-idm-${USER_DIRECTORY}-config.xml
+    echo_info "End Deploying Directory ${USER_DIRECTORY} conf ..."  
+  fi
+}
+
+do_stop_exo_docker() {
+  echo_info "Stopping Exo docker ..."
+  if ! ${DEPLOY_EXO_DOCKER}; then
+    echo_info "Exo wasn't deployed in docker, skiping exo docker shutdown"
+    return
+  fi
+  ensure_docker_container_stopped ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME}
+  echo_info "Exo docker container ${DEPLOYMENT_EXO_DOCKER_CONTAINER_NAME} stopped."
+}
+
+do_start_exo_docker() {
+  local DOCKER_ARGS=""
+  local DATABASE_TYPE=""
+  DEPLOYMENT_LOG_PATH="${DEPLOYMENT_DIR}/logs/exo"
+  if [ -e "${DEPLOYMENT_LOG_PATH}/platform.log" ]; then
+      rm -rf ${DEPLOYMENT_LOG_PATH}/platform.log
+  fi
+  echo_info "Starting Exo docker..."
+  if ! ${DEPLOY_EXO_DOCKER}; then
+    echo_info "Exo wasn't deployed in docker, skiping exo docker startup"
+    return
+  fi
+  mkdir -p ${DEPLOYMENT_DIR}/logs/exo
```

Comment:
The owner or the permissions of the directory have to be adapted to let the exo user inside the container to write